syft/pkg/cataloger/nix/parse_nix_store_path_test.go

```diff
@@ -0,0 +1,284 @@
+package nix
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_findVersionIsh(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		wantIdx int
+		wantEx  string
+		wantPr  string
```

Comment:
Is there a better name for this?

Reply:
yeah, let me rename all of these (ex = version, pr = prerelease)